core/src/main/scala/org/apache/spark/sql/crossdata/execution/commands.scala

```diff
@@ -21,35 +21,75 @@ import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.crossdata.XDContext
 import org.apache.spark.sql.crossdata.catalog.XDCatalog
 import XDCatalog.CrossdataTable
-import org.apache.spark.sql.crossdata.util.CreateRelationUtil
+import org.apache.spark.sql.crossdata.util.CreateRelationUtil.createLogicalRelation
 import org.apache.spark.sql.execution.RunnableCommand
 import org.apache.spark.sql.execution.datasources.{LogicalRelation, ResolvedDataSource}
 import org.apache.spark.sql.sources.{HadoopFsRelation, InsertableRelation}
 import org.apache.spark.sql.types.StructType
 import org.apache.spark.sql.{AnalysisException, DataFrame, Row, SQLContext, SaveMode}
 
-private[crossdata] case class PersistDataSourceTable(
-                                   tableIdent: TableIdentifier,
-                                   userSpecifiedSchema: Option[StructType],
-                                   provider: String,
-                                   options: Map[String, String],
-                                   allowExisting: Boolean) extends RunnableCommand {
+private[crossdata] trait DoCatalogDataSourceTable extends RunnableCommand {
```

Comment:
I just recalled it isn't worth of it since its child are case classes and I'd have to replicate the parameters. In this case its better to use traits.